### ISSUE TITLE
Accessing actions from custom middleware

### ISSUE THREAD

**Issue by dash-** [ericelliott/redux-dsm#19]
Working on an example app, I'm wondering how I should best access a specific action? (Like from `const action = new dsm(...).actions()`).  They could be hard-coded, but that seems like a fragile anti-pattern.  I'd think there would be a way of using my original states (`initialize`, `fetch`, etc) but there doesn't really seem to be a good way to do this.  If `actions` was exposed as a map, it could be `actions.initialize` or `actions.fetch`, for example.

Thoughts?

Code:

```
const dsm = require('redux-dsm');

const fetchingStates = [
  ['initialize', 'idle',
    ['fetch', 'fetching',
      ['cancel', 'idle'],
      ['report error', 'error',
        ['handle error', 'idle']
      ],
      ['report success', 'success',
        ['handle success', 'idle']
      ]
    ]
  ]
];

const myDsm = dsm({
  component: 'myComponent',
  description: 'fetch foo',
  actionStates: fetchingStates
});

const actions = myDsm.actions;

const customMiddleware = store => next => action => {
```
Assuming I want to handle a specific action, how do I access it?
```
}```

**Comment by ericelliott** [comment]
You mean we should expose action type constants as an object with original labels as keys? That's not a bad idea. PR?

**Comment by bognix** [comment]
I would like to propose an alternative

What user provides is this array of states, actions and transitions:
```const fetchingStates = [
  ['initialize', 'idle',
    ['fetch', 'fetching',
      ['cancel', 'idle'],
      ['report error', 'error',
        ['handle error', 'idle']
      ],
      ['report success', 'success',
        ['handle success', 'idle']
      ]
    ]
  ]
];
````

and I like how the `actionCreators` maps state transitions to methods.

If I understand correctly what you are proposing David is to have the same structure in `actions` so calling `actions.handleSuccess` would return string value.

How about combining those two and instead of having both `actionCreators` and `actions` do something like this:
```actions: {
    handleSuccess: {
        creator: () => {},
        name: "NAME"
    },
    ...
}
```
 
On the other hand, I don't like adding another level of nesting and it seems reasonable to just import `{ handleSuccess, handleError} from './myActionCreators' ` and then simply call it without accessing nested `creator`. With this in mind I'm leaning toward David's idea which doesn't include additional nesting but I'm posting above as the food for thought.

**Comment by dash-** [comment]
@ericelliott https://github.com/ericelliott/redux-dsm/pull/20

**Comment by ericelliott** [comment]
Merged. Will look at releasing the change with a major version bump soon.

**Comment by dash-** [comment]
Awesome thanks :-)

**Comment by ericelliott** [comment]
Published as 2.0.0
